### PR TITLE
Prevent units that are not suitable from being used in convertunits

### LINKED ISSUE #27458: ConvertUnits Documentation
**Original reporter:** [PSI]/[Thomas Mueller]

ConvertUnits incorrectly has Degrees listed as a convertible unit

https://forum.mantidproject.org/t/strange-behavior-of-convertunits/591/5

Changing the documentation is one thing, but also worth considering is that there are unofficially two levels of units: Those that are TOF convertible and those that are not. Is having them listed in the same place in the Unit Factory but treated differently (as we currently do) the correct way to be doing things?

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**
Ensures that the target unit validator only lists "convertible units

**To test:**
1. Check that the "fake" units of Empty, Label, Degrees and Temperature are not available in the drop down for ConvertUnits
1. Check that the documentation does not list them as valid options.

Fixes #27458. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
Drop down menu for Convert Units works fine and the list of units on ConvertUnits.html in the docs is now correct. Just need to take them out of the ConvertUnitsUsingDetectorTable algorithm

**Review by DanielMurphy22 — DISMISSED** [review]

**Comment by DanielMurphy22** [comment]
Looks good!

File: Framework/Kernel/inc/MantidKernel/UnitFactory.h [hunk 1]
```diff
@@ -58,6 +54,21 @@ class MANTID_KERNEL_DLL UnitFactoryImpl final : public DynamicFactory<Unit> {
   UnitFactoryImpl(const UnitFactoryImpl &) = delete;
   UnitFactoryImpl &operator=(const UnitFactoryImpl &) = delete;
 
+  /// Returns the names of the convertible units in the factory
+  /// @return A string vector of keys
+  const std::vector<std::string> getConvertibleUnits() const {
+    std::vector<std::string> convertibleUnits;
+
+    for (auto unitKey : getKeys()) {
```
Comment (DanNixon) [inline Framework/Kernel/inc/MantidKernel/UnitFactory.h]:
```suggestion
    for (const auto &unitKey : getKeys()) {
```

File: Framework/Kernel/inc/MantidKernel/UnitFactory.h [hunk 2]
```diff
@@ -58,6 +54,21 @@ class MANTID_KERNEL_DLL UnitFactoryImpl final : public DynamicFactory<Unit> {
...
+      auto unit_sptr = create(unitKey);
```
Comment (DanNixon) [inline Framework/Kernel/inc/MantidKernel/UnitFactory.h]:
```suggestion
      const auto unit_sptr = create(unitKey);
```

**Review by DanNixon — CHANGES_REQUESTED** [review]

**Comment by DanNixon** [comment]
Didn't realise we could merge our own unapproved PRs.

Top notch review process...

**Comment by NickDraper** [comment]
On this one the only change was your two suggested changes

**Comment by DanNixon** [comment]
And fixing the shit commit history.

**Pull request merged**